Client spec v1.2: new --language flag, BCP47 to POSIX and other clarifications (#3168)
tldr-pages/tldr: CLIENT-SPECIFICATION.md

# tldr-pages client specification

 - **Current Specification Version:** 1.2

This document contains the official specification for tldr-pages clients. It is _not_ a specification of the format of the pages themselves - only a specification of how a user should be able to interface with an official client.

The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).

> ## Changelog

Reviewer: Since this changelog is going to get longer and longer, I would suggest moving this section at the end of the file, referencing it in the first paragraph, like this (bold text added by me):

> This document contains the official specification for tldr-pages clients. It is _not_ a specification of the format of the pages themselves - only a specification of how a user should be able to interface with an official client. **For a list of previous versions of the specification, see the [changelog section](#Changelog) below.**

I would also suggest to link to specific versions of the file along with PR numbers, and have the list in reverse order (most recent first), like this (see source of this comment for the Markdown):

 - [v1.2, July 3rd 2019](https://github.com/tldr-pages/tldr/blob/master/CLIENT-SPECIFICATION.md) (#3168)
   - Added new `-L, --language` recommended command-line argument.

 - [v1.1, April 1st 2019](https://github.com/tldr-pages/tldr/blob/fbdc06b7425f92cc0d4fc9a5cfc5860ef017251e/CLIENT-SPECIFICATION.md) (#2859)
   - Clarified platform section.

 - [v1.0, January 23rd 2019](https://github.com/tldr-pages/tldr/blob/f5be8a2614a455288f26e42953efeb8cb3bc50b0/CLIENT-SPECIFICATION.md) (#2706)
   - Initial release.

Author: Done :D

### Arguments

A number of command-line options MUST be supported (unless otherwise specified) if a CLI is implemented:

Option             | Required?   | Meaning
-------------------|-------------|----------
`-v`, `--version`  | Yes         | Shows the current version of the client, and the version of this specification that it implements.
`-p`, `--platform` | Yes         | Specifies the platform to be used to perform the action (either listing or searching) as an argument. If this option is specified, the selected platform MUST be checked first instead of the current platform as described below.
`-u`, `--update`   | Conditional | Updates the offline cache of pages. MUST be implemented if cache is supported.
`-l`, `--list`     | No          | Lists all the pages in the current platform to the standard output. If the special platform `all` is specified a list of all pages in all platforms MUST be displayed.
`-L`, `--language` | No          | Specifies the preferred language for the page returned. Overrides other language detection mechanisms. See the [language section](#language) for more information.

> Clients MAY choose to onlu implement the short version of an option, ignoring the long form.

Reviewer 2: only

`-p`, `--platform`	| Yes		| Specifies the platform to be used to perform the action (either listing or searching). If this option is specified, the selected platform MUST be checked first instead of the current platform as described below.
`-u`, `--update`	| Conditional	| Updates the offline cache of pages. MUST be implemented if cache is supported.
`-l`, `--list`		| No		| Lists all the pages in the current platform to the standard output. If the special platform `all` is specified a list of all pages in all platforms MUST be displayed.

> `--language`		| Yes		| Specifies the preferred language for the page returned. Overrides the `LANG` environment variable - see the [section on translations](#translations) for more information.

Reviewer 2: Ahh .. `-l` is already taken. Have you thought of choosing some other single letter to have a single letter flag ? I don't know .. maybe -e, -g ?

Author: Ah, good idea. Not sure what to choose that would make sense, actually.

> `--language`		| Yes		| Specifies the preferred language for the page returned. Overrides the `LANG` environment variable - see the [section on translations](#translations) for more information.

Reviewer: ```suggestion
`-L, --language`		| No		| Specifies the preferred language for the page returned. Overrides other language detection mechanisms. See the [language section](#language) for more information.
```

It is possible that due to this page resolution logic, the client may show a page which does not belong to the host platform because a page can reside in `common`, and not be present on the host platform. Clients must not assume that a given command is always executable on the host platform.

> #### If a page isn not found

Reviewer 2: is
